snappy_wrappers/wrappers/pvactools/combining/wrapper.py

```diff
@@ -0,0 +1,109 @@
+
+import os
+
+from snakemake.shell import shell
+
+__author__ = "Pham Gia Cuong"
+__email__ = "pham.gia-cuong@bih-charite.de"
+
+step = snakemake.config["pipeline_step"]["name"]
+config = snakemake.config["step_config"][step]
+
+exp_format = config["preparation"]["format"]
+preparation_vcf = os.path.join(
+    os.path.dirname(__file__),
+    "comb_rna.py",
+)
+ensemble_id = (
+    "--ignore-ensembl-id-version" if snakemake.params.args["ignore_ensembl_id_version"] else ""
+)
+max_depth = snakemake.params.args["max_depth"]
+format = snakemake.params.args["format"]
+mode = snakemake.params.args["mode"]
+expression_column = (
+    f"--expression-file {snakemake.params.args['expression_column']}" if format == "custom" else ""
+)
+
+id_column = f"--expression-file {snakemake.params.args['id_column']}" if format == "custom" else ""
+
+bam_file = snakemake.input.bam
+gencount_file = snakemake.input.expression
+
+rna_vcf = (
+    f"--rna-vcf $TMPDIR/{snakemake.wildcards.tumor_library}.tmp.vcf.gz "
+    if format == "snappy_custom"
+    else ""
+)
+expression_file = (
+    f"--expression-file {gencount_file} "
+    if format == "snappy_custom"
+    else config["preparation"]["expression_file"]
+)
+
+shell(
+    r"""
+# -----------------------------------------------------------------------------
+# Redirect stderr to log file by default and enable printing executed commands
+# Also pipe stderr to log file
+if [[ -n "{snakemake.log.log}" ]]; then
+    if [[ "$(set +e; tty; set -e)" != "" ]]; then
+        rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+        exec 2> >(tee -a "{snakemake.log.log}" >&2)
+    else
+        rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+        echo "No tty, logging disabled" >"{snakemake.log.log}"
+    fi
+fi
+# -----------------------------------------------------------------------------
+# Create auto-cleaned temporary directory
+export TMPDIR=$(mktemp -d)
+trap "rm -rf $TMPDIR" EXIT
+
+set -x
+# -----------------------------------------------------------------------------
+
+# Write out information about conda installation
+conda list > {snakemake.log.conda_list}
+conda info > {snakemake.log.conda_info}
+
+if [[ {format}=="snappy_custom" ]]; then
+    # Getting region of SNVs only
+    bcftools filter -i 'TYPE="snp"' {snakemake.input.vcf} | bcftools query -f '%CHROM\t%POS0\t%END\n' > $TMPDIR/{snakemake.wildcards.tumor_library}.tmp.bed
+
+    # Getting snvs from RNA sequencing data
+    bcftools mpileup -Ov \
+        --annotate FORMAT/AD,FORMAT/DP \
+        -R $TMPDIR/{snakemake.wildcards.tumor_library}.tmp.bed \
+        -f {snakemake.config[static_data_config][reference][path]} \
+        --per-sample-mF \
+        --max-depth {max_depth} \
+        --redo-BAQ -Oz \
+        -o $TMPDIR/{snakemake.wildcards.tumor_library}.tmp.vcf.gz \
+        {bam_file}
+    tabix -f $TMPDIR/{snakemake.wildcards.tumor_library}.tmp.vcf.gz
+fi
+
+#Need to add option for RNA vcf as well
```

Comment:
Ideally, we'd like to split this wrapper into two parts: the pileup & your python program. These two should be connected with a temp file, now that Snakemake allows using them between rules.